src/ffi.rs

```diff
@@ -142,22 +150,379 @@ extern "C" {
+    }
+
+    /// Sets WKB output dimensions.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, Dimensions};
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    ///
+    /// context_handle.set_WKB_output_dimensions(Dimensions::TwoD);
+    /// assert!(context_handle.get_WKB_output_dimensions() == Dimensions::TwoD);
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn set_WKB_output_dimensions(&self, dimensions: Dimensions) -> Dimensions {
+        Dimensions::from(unsafe { GEOS_setWKBOutputDims_r(self.ptr, dimensions.into()) })
+    }
+
+    /// Gets WKB byte order.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, ByteOrder};
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    ///
+    /// context_handle.set_WKB_byte_order(ByteOrder::LittleEndian);
+    /// assert!(context_handle.get_WKB_byte_order() == ByteOrder::LittleEndian);
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn get_WKB_byte_order(&self) -> ByteOrder {
+        ByteOrder::from(unsafe { GEOS_getWKBByteOrder_r(self.ptr) })
+    }
+
+    /// Sets WKB byte order.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, ByteOrder};
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    ///
+    /// context_handle.set_WKB_byte_order(ByteOrder::LittleEndian);
+    /// assert!(context_handle.get_WKB_byte_order() == ByteOrder::LittleEndian);
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn set_WKB_byte_order(&self, byte_order: ByteOrder) -> ByteOrder {
+        ByteOrder::from(unsafe { GEOS_setWKBByteOrder_r(self.ptr, byte_order.into()) })
+    }
+
+    /// Convert [`GGeom`] from WKB format.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, GGeom};
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    /// let point_geom = GGeom::new("POINT (2.5 2.5)").expect("Invalid geometry");
+    /// let wkb_buf = context_handle.geom_to_WKB_buf(&point_geom)
+    ///                             .expect("conversion to WKB failed");
+    /// let new_geom = context_handle.geom_from_WKB_buf(wkb_buf.as_ref())
+    ///                              .expect("conversion to WKB failed");
+    /// assert!(point_geom.equals(&new_geom) == Ok(true));
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn geom_from_WKB_buf(&self, wkb: &[u8]) -> GeosResult<GGeom> {
```

Comment:
I just thought: should I take a `String` as parameter in here?

Reply:
WKB is binary, not utf-8, so `&[u8]` is your only solution.

Reply:
Why WKB in caps? Why not `geom_from_wkb_buf` as used in the rust convension? https://doc.rust-lang.org/nightly/std/str/fn.from_utf8.html

Reply:
I can update the name. :)

Reply:
Ah right: I mixed up the name with WKT.